src/client/api/event.c

```diff
@@ -328,8 +328,6 @@ daos_event_complete_locked(struct daos_eq_private *eqx,
 
 	if (eqx != NULL)
 		eq = daos_eqx2eq(eqx);
-	else
-		D_MUTEX_LOCK(&evx->evx_lock);
```

Comment:
not a dumb question, but yes if the parent event is in an eq, the eq is locked. however if the parent is independent, there is no lock, but i don't think a lock is necessary here because i believe only 1 child event completion can trigger the parent completion part. i can look deeper on this but probably can do it in a separate PR if a fix is necessary for that. in any case, we really have no use case at all for the parent event and i think at some point probably it's better to just remove it. it was initially added before we had tse and dependencies.

Reply:
Ah, glad I asked; the answer is deeper than I anticipated. Thanks, Mohamad.